### ISSUE TITLE
Create an official docker image of the project and host it on Docker Hub.

### ISSUE THREAD

**Issue by alextselegidis** [alextselegidis/easyappointments#1116]
<small>
<table>
<tr>
<td><img src="https://easyappointments.org/img/easyappointments-signature.png"></td>
<td>&nbsp;&nbsp;Alex Tselegidis, Easy!Appointments Creator
<br>&nbsp;&nbsp;<a href="https://easyappointments.org/get-a-free-quote">Need a customization? Get a free quote!</a></td>
</tr>
</table>
</small>

**Comment by andrewclausen** [comment]
You might find this project helpful:

<https://github.com/clayrisser/easyappointments-docker>

It worked for me, even though it hasn't been updated in 4 years.

**Comment by alextselegidis** [comment]
Thanks @andrewclausen, that might actually be a good place to start!

**Comment by JanC** [comment]
hey, what's the status of the Docker support?

I tried building and running a locally using the [docker-compose.yml](https://github.com/alextselegidis/easyappointments/tree/master/docker)

The services start but I get an error when accessing http://localhost:8000/

```
Warning: require_once(/var/www/html/config.php): failed to open stream: No such file or directory in /var/www/html/index.php on line 50

Fatal error: require_once(): Failed opening required '/var/www/html/config.php' (include_path='.:/usr/local/lib/php') in /var/www/html/index.php on line 50
```

**Comment by alextselegidis** [comment]
Hi @JanC Please copy config-sample.php to config.php and add your installation information inside. 

You can follow this guide here: 

https://easyappointments.org/docs.html#1.4.2/docker.md

<small>
<table style">
<tr>
<td><img src="https://easyappointments.org/img/easyappointments-signature.png" style="margin-right: 5px"></td>
<td style="line-height: 2">Alex Tselegidis, Easy!Appointments Creator <br><a href="https://easyappointments.org/get-a-free-quote">Need a customization? Get a free quote!</a></td>
</tr>
</table>
</small>

**Comment by alextselegidis** [comment]
Hi @lambchop01, 

You will need to run the `composer install` command on your host machine before starting the containers. 

<small>
<table>
<tr>
<td><img src="https://easyappointments.org/img/easyappointments-signature.png" style="margin-right: 5px"></td>
<td style="line-height: 2">Alex Tselegidis, Easy!Appointments Creator <br><a href="https://easyappointments.org/get-a-free-quote">Need a customization? Get a free quote!</a></td>
</tr>
</table>
</small>

**Comment by lambchop01** [comment]
@alextselegidis ,
Fair! I was thinking that was part of the docker container build process, but actually looked and see that it is not.  I'll give that a shot when I get a chance. 
Thanks

**Comment by dukeofgaming** [comment]
I get this while trying composer update:

```
 Problem 1
    - Root composer.json requires PHP extension ext-gd * but it is missing from your system. Install or enable PHP's gd extension.
  Problem 2
    - gregwar/captcha[v1.1, ..., v1.1.9] require ext-gd * -> it is missing from your system. Install or enable PHP's gd extension.
    - Root composer.json requires gregwar/captcha ^1.1 -> satisfiable by gregwar/captcha[v1.1, ..., v1.1.9].

To enable extensions, verify that they are enabled in your .ini files:
    - C:\tools\php80\php.ini
```

It would sure be nice to have a prebuilt image, it's too much trouble to create a new Dockerfile to repackage the whole environment or change my existing one just to try the app (your demo backend is not working BTW, administrator/administrator won't login)

I'm going to have to pass on trying out your solution and will probably go for setmore or sagenda, which is not my preference since I wanted something 100% self-hosted. I hope you come up with an official image soon.

**Comment by alextselegidis** [comment]
Hi @dukeofgaming,

Thanks for reaching out! 

This ticket has been on my plans for 1.5 but I forgot to assign the milestone properly.

Hold on for some days (if possible) and I will get back here with the official docker image and more documentation. 

Cheers!

<small>
<table>
<tr>
<td><img src="https://easyappointments.org/img/easyappointments-signature.png" style="margin-right: 5px"></td>
<td style="line-height: 2">Alex Tselegidis, Easy!Appointments Creator <br><a href="https://easyappointments.org/get-a-free-quote">Need a customization? Get a free quote!</a></td>
</tr>
</table>
</small>

**Comment by alextselegidis** [comment]
@dukeofgaming FYI the official docker file will be hosted on this repo: 

https://github.com/alextselegidis/easyappointments-docker

Feel free to star and follow and stay up to date with the latest development state. 

<small>
<table>
<tr>
<td><img src="https://easyappointments.org/img/easyappointments-signature.png" style="margin-right: 5px"></td>
<td style="line-height: 2">Alex Tselegidis, Easy!Appointments Creator <br><a href="https://easyappointments.org/get-a-free-quote">Need a customization? Get a free quote!</a></td>
</tr>
</table>
</small>

**Comment by noraj** [comment]
This docker-compose works too https://github.com/stw-on/easyappointments

**Comment by alextselegidis** [comment]
Hello @noraj, thanks for sharing this! 

I'll check this out, although it looks like this setup is for an older Easy!Appointments version. 

<small>
<table>
<tr>
<td><img src="https://easyappointments.org/img/easyappointments-signature.png" style="margin-right: 5px"></td>
<td style="line-height: 2">Alex Tselegidis, Easy!Appointments Creator <br><a href="https://easyappointments.org/get-a-free-quote">Need a customization? Get a free quote!</a></td>
</tr>
</table>
</small>

**Comment by noraj** [comment]
> I'll check this out, although it looks like this setup is for an older Easy!Appointments version.

Yes they stopped at 1.3.2, hardcoded some value, not everything is clean there but at least it works.

**Comment by alextselegidis** [comment]
@noraj I'm currently working on an image that contains PHP-fpm with Nginx on Alpine, which makes it pretty lightweight. The best would be to provide various flavors (with apache, cli etc) but I think for now I will just publish one version and the rest will come at some point later. 

<small>
<table>
<tr>
<td><img src="https://easyappointments.org/img/easyappointments-signature.png" style="margin-right: 5px"></td>
<td style="line-height: 2">Alex Tselegidis, Easy!Appointments Creator <br><a href="https://easyappointments.org/get-a-free-quote">Need a customization? Get a free quote!</a></td>
</tr>
</table>
</small>

**Comment by Typhonragewind** [comment]
> @noraj I'm currently working on an image that contains PHP-fpm with Nginx on Alpine, which makes it pretty lightweight. The best would be to provide various flavors (with apache, cli etc) but I think for now I will just publish one version and the rest will come at some point later.
> ![](https://camo.githubusercontent.com/30676c4987784d10e020c434b4519580e70da2388d2b1032698121af407caeff/68747470733a2f2f656173796170706f696e746d656e74732e6f72672f696d672f656173796170706f696e746d656e74732d7369676e61747572652e706e67) 	Alex Tselegidis, Easy!Appointments Creator
> [Need a customization? Get a free quote!](https://easyappointments.org/get-a-free-quote)

This would be amazing!
Thanks for this great software :D

**Comment by riccardomessineo** [comment]
Any news about the official docker image?

**Comment by alextselegidis** [comment]
Hello @riccardomessineo, currently finalizing, I'll report here in a couple of days.

<small>
<table>
<tr>
<td><img src="https://easyappointments.org/img/easyappointments-signature.png" style="margin-right: 5px"></td>
<td style="line-height: 2">Alex Tselegidis, Easy!Appointments Creator <br><a href="https://easyappointments.org/get-a-free-quote">Need a customization? Get a free quote!</a></td>
</tr>
</table>
</small>

**Comment by alextselegidis** [comment]
Hello! 

I just deployed the first official image of Easy!Appointments!!! 

Currently it has the latest stable release (1.4.3) but shortly the 1.5.0 will follow. 

<small>
<table>
<tr>
<td><img src="https://easyappointments.org/img/easyappointments-signature.png" style="margin-right: 5px"></td>
<td style="line-height: 2">Alex Tselegidis, Easy!Appointments Creator <br><a href="https://easyappointments.org/get-a-free-quote">Need a customization? Get a free quote!</a></td>
</tr>
</table>
</small>
